libs/database/src/collab/collab_db_ops.rs

```diff
@@ -55,109 +55,64 @@ pub async fn insert_into_af_collab(
-          existing_workspace_id,
-          workspace_id
-        )));
-      }
-    },
-    None => {
-      // If the collab doesn't exist, insert a new row into the `af_collab` table and add a corresponding
-      // entry to the `af_collab_member` table.
-      let permission_id: i32 = sqlx::query_scalar!(
-        r#"
-          SELECT rp.permission_id
-          FROM af_role_permissions rp
-          JOIN af_roles ON rp.role_id = af_roles.id
-          WHERE af_roles.name = 'Owner';
-        "#
-      )
-      .fetch_one(tx.deref_mut())
-      .await?;
-
-      sqlx::query!(
-        r#"
-        INSERT INTO af_collab_member (uid, oid, permission_id)
-        VALUES ($1, $2, $3)
-        ON CONFLICT (uid, oid)
-        DO UPDATE
-          SET permission_id = excluded.permission_id;
-        "#,
-        uid,
-        params.object_id,
-        permission_id
-      )
-      .execute(tx.deref_mut())
-      .await
-      .map_err(|err| {
-        AppError::Internal(anyhow!(
-          "Insert af_collab_member failed: {}:{}:{}. error details:{:?}",
-          uid,
-          params.object_id,
-          permission_id,
-          err
-        ))
-      })?;
+  sqlx::query!(
+    r#"
+  INSERT INTO af_collab (oid, blob, len, partition_key, encrypt, owner_uid, workspace_id)
+  VALUES ($1, $2, $3, $4, $5, $6, $7)
+  ON CONFLICT (oid, partition_key) DO UPDATE
+  SET blob = $2, len = $3, encrypt = $5, owner_uid = $6;
+  "#,
+    params.object_id,
+    params.encoded_collab_v1.as_ref(),
+    params.encoded_collab_v1.len() as i32,
+    partition_key,
+    encrypt,
+    uid,
+    workspace_id,
+  )
+  .execute(tx.deref_mut())
+  .await
+  .map_err(|err| {
+    AppError::Internal(anyhow!(
+      "Update af_collab failed: workspace_id:{}, uid:{}, object_id:{}, collab_type:{}. error: {:?}",
+      workspace_id,
+      uid,
+      params.object_id,
+      params.collab_type,
+      err,
+    ))
+  })?;
 
-      sqlx::query!(
-        "INSERT INTO af_collab (oid, blob, len, partition_key, encrypt, owner_uid, workspace_id)\
-          VALUES ($1, $2, $3, $4, $5, $6, $7)",
-        params.object_id,
-        params.encoded_collab_v1.as_ref(),
-        params.encoded_collab_v1.len() as i32,
-        partition_key,
-        encrypt,
-        uid,
-        workspace_id,
-      )
-      .execute(tx.deref_mut())
-      .await.map_err(|err| {
-        AppError::Internal(anyhow!(
-          "Insert new af_collab failed: workspace_id:{}, uid:{}, object_id:{}, collab_type:{}. payload len:{} error: {:?}",
-         workspace_id, uid, params.object_id, params.collab_type, params.encoded_collab_v1.len(), err,
-        ))
-      })?;
-    },
-  }
+  sqlx::query!(
+    r#"
+  INSERT INTO af_collab_member (uid, oid, permission_id)
+  VALUES ($1, $2, (SELECT rp.permission_id 
+    FROM af_role_permissions rp
+    JOIN af_roles ON rp.role_id = af_roles.id
+    WHERE af_roles.name = 'Owner'))
+  ON CONFLICT (uid, oid)
```

Comment:
You don't need to separately select then insert af_collab_member permissions. It can be done in one go.